src/sentry/api/endpoints/organization_environments.py

```diff
@@ -0,0 +1,23 @@
+from __future__ import absolute_import
+
+from rest_framework.response import Response
+
+from sentry.api.bases import OrganizationEndpoint
+from sentry.api.serializers import serialize
+from sentry.models import Environment
+
+
+class OrganizationEnvironmentsEndpoint(OrganizationEndpoint):
+    def get(self, request, organization):
+        queryset = Environment.objects.filter(
+            organization_id=organization.id,
+        ).exclude(
+            # HACK(mattrobenolt): We don't want to surface the
+            # "No Environment" environment to the UI since it
+            # doesn't really exist. This might very likely change
+            # with new tagstore backend in the future, but until
+            # then, we're hiding it since it causes more problems
+            # than it's worth.
+            name='',
+        ).order_by('name')
+        return Response(serialize(list(queryset), request.user))
```

Comment:
I don't remember what the 90th+ %ile is here but it's pretty high, IIRC. Not sure if that matters here.

Reply:
@billyvg I assume we wouldn't want to paginate here? I don't know any situation where you'd only want some of them?

I don't mind paginating, but we would also currently have this issue for the `EnvironmentProject` endpoint as well since that's basically the same 90p. Just assuming you'd then just be making multiple API calls to fetch them all.

Reply:
I think pagination is fine if we're able to search for an environment

Reply:
Let's circle back on this if it becomes an issue.